packages/types/src/documents/app/view.ts

```diff
@@ -33,15 +33,24 @@ export interface View {
   groupBy?: string
 }
 
-export type ViewFieldMetadata = UIFieldMetadata & {
+export interface BasicViewUIFieldMetadata extends UIFieldMetadata {
   readonly?: boolean
   columns?: Record<string, RelationSchemaField>
 }
 
-export type RelationSchemaField = UIFieldMetadata & {
+export interface RelationSchemaField extends UIFieldMetadata {
   readonly?: boolean
 }
 
+export interface ViewCalculationFieldMetadata extends BasicViewUIFieldMetadata {
+  calculationType: CalculationType
+  field: string
+}
+
+export type ViewUIFieldMetadata =
```

Comment:
I have a feeling when I started this PR, this type was called `ViewUIFieldMetadata` and somewhere it got changed to `ViewFieldMetadata` but that got lost. I didn't add `UI` to this name in this PR.